File: solutions/_303_range_sum_query_immutable/solution.java
package _303_range_sum_query_immutable;

class NumArray {

  private int[] sums;

  public NumArray(int[] nums) {
    sums = new int[nums.length];
    int sum = 0;
    for (int i = 0; i < nums.length; i++) {
      sum += nums[i];
      sums[i] = sum;
    }
  }

  public int sumRange(int left, int right) {
    return sums[right] - (left > 0 ? sums[left - 1] : 0);
  }
}
/*
https://leetcode.com/problems/range-sum-query-immutable/
*/

File: solutions/_303_range_sum_query_immutable/solution.py
from typing import List


class NumArray:
    def __init__(self, nums: List[int]):
        self.sums = []
        cur = 0
        for n in nums:
            cur += n
            self.sums.append(cur)

    def sumRange(self, left: int, right: int) -> int:
        r = self.sums[right]
        l = self.sums[left - 1] if left > 0 else 0
        return r - l


# https://leetcode.com/problems/range-sum-query-immutable/
